src/toil/wdl/wdltoil.py

```diff
@@ -470,11 +485,21 @@ def _devirtualize_filename(self, filename: str) -> str:
         if filename.startswith(TOIL_URI_SCHEME):
             # This is a reference to the Toil filestore.
             # Deserialize the FileID
-            file_id, file_basename = unpack_toil_uri(filename)
+            file_id, parent_id, file_basename = unpack_toil_uri(filename)
 
             # Decide where it should be put
-            file_dir = self._file_store.getLocalTempDir()
-            dest_path = os.path.join(file_dir, file_basename)
+            if self._is_uuid(parent_id):
+                # This is a URI with the "parent" UUID attached to the filename
+                # Use UUID as folder name rather than a new temp folder to reduce internal clutter
+                if not os.path.exists(parent_id):
+                    os.mkdir(parent_id)
```

Comment:
It looks like we implicitly rely on `self._file_store.localTempDir` being the current directory here?

Probably we should construct the path to the directory we want to put the file in, make it, construct the file path from that, and return that.